### PR TITLE
[USM] java ci docker based tests

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Adding some E2E java tests with the tracer
Here an example of injection with JustWait demo
But more tests WgetJava like would be added

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/testutil/patternscanner.go [hunk 1]
```diff
@@ -45,3 +50,11 @@ func (ps *PatternScanner) Write(p []byte) (n int, err error) {
 
 	return
 }
+
+func (ps *PatternScanner) PrintLogs(t *testing.T) {
+	logs := ""
+	for _, str := range ps.buffers {
+		logs += str
+	}
+	t.Log(logs)
```
Comment (hmahmood) [inline pkg/network/protocols/testutil/patternscanner.go:59]: `t.Log(strings.Join(ps.buffers, ""))`

**Review by hmahmood — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

File: pkg/network/tracer/tracer_usm_linux_test.go [hunk 2]
```diff
@@ -589,6 +594,92 @@ func testProtocolClassificationMapCleanup(t *testing.T, cfg *config.Config, clie
...
+		_, err = nettestutil.RunCommand("install -m444 " + dir + "/../java/testdata/TestAgentLoaded.jar " + agentDir + "/agent-usm.jar")
+		require.NoError(t, err)
+	}
+
+	// testContext shares the context of a given test.
+	// It contains common variable used by all tests, and allows extending the context dynamically by setting more
+	// attributes to the `extras` map.
+	type testContext struct {
+		// A channel to mark goroutines (like servers) to halt.
+		done chan struct{}
```
Comment (guyarb) [inline pkg/network/tracer/tracer_usm_linux_test.go:622]: not being used

File: pkg/network/java/testdata/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+version: '3'
+services:
+  java:
+    image: ${IMAGE_VERSION}
+    entrypoint: java -cp /v ${ENTRYCLASS}
+    volumes:
+      - ${TESTDATA_DIR}:/v
```
Comment (guyarb) [inline pkg/network/java/testdata/docker-compose.yml:7]: volumes are relative to docker-compose locations
```suggestion
    volumes:
      - ./:/v
```

File: pkg/network/java/testdata/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+version: '3'
```
Comment (guyarb) [inline pkg/network/java/testdata/docker-compose.yml:1]: you should add the path of docker-compose to `save_test_dockers` command in `system-probe.py` in order to pull the image in the CI, and especially the images that you want to pull..

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/protocols/testutil/patternscanner.go (hunk 1 above)
Comment (nplanel) [reply pkg/network/protocols/testutil/patternscanner.go]: thanks

File: pkg/network/java/testdata/docker-compose.yml (hunk 3 above)
Comment (nplanel) [reply pkg/network/java/testdata/docker-compose.yml]: done

**Review by guyarb — APPROVED** [review]

**Pull request merged**